### PR TITLE
refactor(mpz-share-conversion): new impl

### PR DESCRIPTION
**Pull request by sinui0**
This PR stubs out the traits for `mpz-share-conversion` to accomodate #112

### CODE REVIEW

**Comment by sinui0** [comment]
I will use these stubs to move ahead with refactoring TLSNotary

File: crates/mpz-share-conversion/src/lib.rs [hunk 1]
```diff
@@ -1,189 +1,56 @@
...
-/// A channel used by conversion protocols for messaging
-pub type ShareConversionChannel<T> = Box<dyn Duplex<ShareConversionMessage<T>>>;
+pub use error::ShareConversionError;
 
 /// A trait for converting additive shares into multiplicative shares.
 #[async_trait]
-pub trait AdditiveToMultiplicative<T: Field> {
-    /// Converts additive shares into multiplicative shares
-    async fn to_multiplicative(&self, input: Vec<T>) -> Result<Vec<T>, ShareConversionError>;
+pub trait AdditiveToMultiplicative<Ctx, T> {
```
Comment (th4s) [inline crates/mpz-share-conversion/src/lib.rs:21]: No trait bounds for field, `pub trait AdditiveToMultiplicative<Ctx, F: Field>`?

**Review by th4s — APPROVED** [review]
:+1:

**Comment by sinui0** [comment]
Will rebase onto #147 prior to merge.

**Review by themighty1 — APPROVED** [review]

**Pull request merged**